### PR TITLE
Extend copyright present analyzer with year and company name

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
And make company name configurable in .editorconfig (see our own .editorconfig)

### CODE REVIEW

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs [hunk 1]
```diff
@@ -76,5 +81,22 @@ private static void CreateDiagnostic(SyntaxNodeAnalysisContext context, Location
...
 			context.ReportDiagnostic(diagnostic);
 		}
+
+		private static bool CheckCopyrightStatement(SyntaxNodeAnalysisContext context, SyntaxTrivia trivia) {
+			var comment = trivia.ToFullString();
+			// Check the copyright mar itself
+			var hasCopyright = comment.Contains("©") || comment.Contains("Copyright");
+			
+			// Check the year
+			var yearRegex = new Regex(@"\d\d\d\d");
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs:91]: A Regex here feels like a relatively expensive operation to run for every compilation unit, but I don't have data to back up this concern of a performance hit.

**Review by bcollamore — APPROVED** [review]

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs]: Made the RegEx a static variable. This at least save the RegEx compile time.

Had to put a lock around it, as a save guard during concurrent execution.

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs [hunk 2]
```diff
@@ -76,5 +82,25 @@ private static void CreateDiagnostic(SyntaxNodeAnalysisContext context, Location
...
 		}
+
+		private static bool CheckCopyrightStatement(SyntaxNodeAnalysisContext context, SyntaxTrivia trivia) {
+			var comment = trivia.ToFullString();
+			// Check the copyright mar itself
+			var hasCopyright = comment.Contains("©") || comment.Contains("Copyright");
+			
+			// Check the year
+			bool hasYear;
+			lock (yearRegex)
```
Comment (mayerj) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs:93]: You don't need this, Regex objects are thread-safe for just calling something like Match.

Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/CopyrightPresentAnalyzer.cs]: You're right, removed the locks again.

**Review by bcollamore — APPROVED** [review]

**Pull request merged**